traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -103,62 +89,136 @@ func UpdateHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Validate parameters before attempting to apply them (don't want to partially apply various status before an error)
+	var configUpdateTime time.Time
+	if hasConfig_update_time {
+		configUpdateTime, err = time.Parse(time.RFC3339Nano, config_update_time)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'config_update_time' must be valid RFC3339 format"), nil)
+			return
+		}
+	}
+
+	var configApplyTime time.Time
+	if hasConfig_apply_time {
+		configApplyTime, err = time.Parse(time.RFC3339Nano, config_apply_time)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'config_apply_time' must be valid RFC3339 format"), nil)
+			return
+		}
+	}
+
+	var revalUpdateTime time.Time
+	if hasRevalidate_update_time {
+		revalUpdateTime, err = time.Parse(time.RFC3339Nano, revalidate_update_time)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'revalidate_update_time' must be valid RFC3339 format"), nil)
+			return
+		}
+	}
+
+	var revalApplyTime time.Time
+	if hasRevalidate_apply_time {
+		revalApplyTime, err = time.Parse(time.RFC3339Nano, revalidate_apply_time)
+		if err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("query parameter 'revalidate_apply_time' must be valid RFC3339 format"), nil)
+			return
+		}
+	}
+
+	if hasUpdated && (hasConfig_update_time || hasConfig_apply_time) {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("potential conflict between params. use 'config_update_time' or 'config_apply_time' over 'updated'"), nil)
+		return
+	}
+
+	if hasRevalUpdated && (hasRevalidate_update_time || hasRevalidate_apply_time) {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("potential conflict between params. use 'reval_update_time' or 'reval_apply_time' over 'reval_updated'"), nil)
+		return
+	}
+
 	strToBool := func(s string) bool {
-		return !strings.HasPrefix(strings.ToLower(s), "f")
+		return !strings.HasPrefix(s, "f")
 	}
 
-	updatedPtr := (*bool)(nil)
 	if hasUpdated {
 		updatedBool := strToBool(updated)
-		updatedPtr = &updatedBool
+		if updatedBool {
+			if err = dbhelpers.QueueUpdateForServer(inf.Tx.Tx, int64(serverID)); err != nil {
```

Comment:
nit: possible truncation should check bounds. I'm not going to repeat that for everywhere I see the cast, but if you do change it, it should change everywhere (probably just want to cast it only once so you can check it only once)